### DISCUSSION TITLE
Weight parameter of DiceLoss

### DISCUSSION THREAD

**Discussion by NastaranVB** [Project-MONAI/MONAI#7606] [Q&A]
Hello,
 I have used loss function as below:
lloss_function=DiceLoss(include_background=False, to_onehot_y=True, softmax=True, reduction='mean', smooth_nr=0, smooth_dr=1e-5, weight=[2, 10])
In past days it was working but now I get the below error:

TypeError                                 Traceback (most recent call last)
/tmp/ipykernel_1144963/2523664510.py in <module>
     11                 model=unet,
     12                 learning_rate=1e-3,
---> 13                 loss_function=DiceLoss(include_background=False, to_onehot_y=True, softmax=True, reduction='mean', smooth_nr=0, smooth_dr=1e-5, weight=[2, 10]),
     14                 optimizer_name=torch.optim.AdamW,
     15                 scheduler_name=torch.optim.lr_scheduler.ExponentialLR,

TypeError: __init__() got an unexpected keyword argument 'weight'

**Comment by KumoLiu** [comment]
Hi @NastaranVB, please check the type of `DiceLoss`. Thanks.

**Reply by NastaranVB** [reply]
Hi @KumoLiu  thanks for coming back to me so quickly. What do you mean from the type of DiceLoss?
I have used it as loss function in multi class segmentation which includes 3 class including background.
'DiceLoss
class monai.losses.DiceLoss(include_background=True, to_onehot_y=False, sigmoid=False, softmax=False, other_act=None, squared_pred=False, jaccard=False, reduction=LossReduction.MEAN, smooth_nr=1e-05, smooth_dr=1e-05, batch=False, weight=None)'

**Reply by KumoLiu** [reply]
Can you check your MONAI version? weight args is added after 1.2. Thanks.
<img width="1281" alt="Screen Shot 2024-04-03 at 23 45 37" src="https://github.com/Project-MONAI/MONAI/assets/55491388/f383b5ef-c7ce-46da-b47c-0a7e01484317">

**Comment by NastaranVB** [comment]
Thank you very much. The version was the reason.
